### ISSUE TITLE
widgets not always visible

### ISSUE THREAD

**Issue by veitveit** [ProtProtocols/IsoLabeledProtocol#69]
Sometimes, even when starting a new image, the widgets do not show. This could be due to already available files in the mapped folder, reloading the notebook, or others.

I see 2 simple possibilities to solve this:
- a fixed button (but needs to show all the time) to reset interface
- some text that tells "run the cell below by pressing shift-return"

@jgriss What do you think?

**Comment by jgriss** [comment]
@veitveit In my tests I found that this happens if the Kernel is already running and the notebook is being reopened. Can you confirm this?

**Comment by jgriss** [comment]
@veitveit After several tests I can reproduce the error consistently by closing the notebook, leaving the kernel running and then opening the notebook again.

it seems that in this case, the `init_cell` script is reexecuting the code (as it should) but the python kernel is doing so as well. This seems to lead to ?race conditions which causes the output (ie. all GUI elements) to be created but then deleted again and not re-created.

An ideal solution would probably be to block the `init_cell` routine if the GUI elements are already present. But this will require some coding...

**Comment by jgriss** [comment]
@veitveit I've now removed all init_cells but one. This displays a button to trigger the workflow and initializes all other cells. When reopening a notebook connected to a running kernel, the message box that settings were reloaded is displayed twice (no idea why) but the user interface remains visible.

Please test!

**Comment by veitveit** [comment]
@jgriss 

And when closing and opening the notebook while running a database search, then I don't get any button. Instead I get the text in the cell:
```
Button(description='Start Wokflow / Display user interface', layout=Layout(width='30%'), style=ButtonStyle())
```

Then, after the search is finished, the widgets show again, but not the button you first implemented.

This looks very tricky and probably is dependent on the browser + OS. Your adjustments seem to work and we cannot change the behavior while a cell is running. 

I suggest to additionally write that the user should close the notebook and open it again as soon as it finished.

**Comment by veitveit** [comment]
@jgriss  What works very well is opening the protocol from a folder which already contains an OUT subfolder with results. Then only the new button shows, so it definitely is needed.

**Comment by jgriss** [comment]
@veitveit 

There are a few issues you mentioned here:
  1. The hide_code widget is no longer supported for the Jupyter version we use. We'd have to change to different widget that basically offers the same functionality. So this is rather separate.
  2. If you open the Notebook while a search is running the following happens:
      - the search is actually running as part of the Python kernel. So the python kernel is busy.
      - The notebook is waiting for the python kernel to finish / be ready to render new elements
      - If this waiting period is causing a timeout the code simply is not rendered properly
      - So this is caused by Jupyter notebook's design, there's nothing we can change about that. This partly refers to #52 where I am thinking about ways we can display the search progress / output even though the notebook wasn't open. Currently, this output is lost and there is no way to get it.

So in short, these two scenarios are not really linked to our original issue. I suggest that we
  * add a text telling users not to close the notebook while the search is running
  * replace the hide_code widget

**Comment by veitveit** [comment]
Yes, adding a bit of text will be the solution for now. 

The hide_code extension still seems to be maintained, but slowly. So it might be supported again.
If there is a better option, let's go for it. I just didn't see anything when I was looking for it.

**Comment by veitveit** [comment]
Seems to work fine now.
